### PR TITLE
Get-Member-related koans need to be explained better

### LINKED ISSUE #379: Get-Member-related koans need to be explained better
# Describe "Koan Bug, Issue, or Help Request"

There are examples in AboutDiscovery as well as AboutGetMember; both could use some improvements and/or rewording.

### AboutDiscovery

https://github.com/vexx32/PSKoans/blob/6c8c91a5958683c08387b4d03ac19b695d476b35/PSKoans/Koans/Cmdlets%201/AboutDiscovery.Koans.ps1#L101-L130

The koan here expects a user to:

1. Look at the list of method names (using `Get-Member`) -- a syntax example of how to use Get-Member to get started is probably a good idea here.
2. Pick the right method name -- most users unfamiliar will likely need to use the documentation; perhaps a handy link to the documentation for `System.String` would be useful here.
3. Supply the correct arguments for the method.

The presentation of the koan could also be problematic; the lines that check the answer require a few leaps of knowledge to understand what's going on. Perhaps adding a comment to illustrate what it's doing in the more typical syntax would go a long way here.

e.g.,:

> This sequence of commands takes the input numbers below and calls the named method on each of them in turn, so it's kind of like doing this: `"1".Foo($MethodArguments)` if "Foo" was the method name you chose.

### AboutGetMember

https://github.com/vexx32/PSKoans/blob/6c8c91a5958683c08387b4d03ac19b695d476b35/PSKoans/Koans/Introduction/AboutGetMember.Koans.ps1#L47-L131

We need to be clearer around what we're asking from the user here. Users not familiar may not be immediately aware of the difference between properties and methods in the Get-Member output, which can be confusing.

Also, some reference to how the documentation shows methods / how to find the documentation is definitely needed here. We can probably also split these out into separate tests, perhaps stepping up the complexity; i.e., first case, just provide the method name (cmdlet name is provided), second case they gotta fill in their own command and method.

A disclaimer here that not all modules will necessarily be available might be a good idea, and some recommendations of how to look through Get-Command to find commands that will be available.

### PR DESCRIPTION
**Pull request by Maamue**
# PR Summary

There are examples in AboutDiscovery as well as AboutGetMember that need a better explanation

Fixes #379 

## Context

Newcomers may not be familiar with the intricaciess of methods and properties. This PR intends to enhance the experience users get when solving the koans, by explaining on a surface level the distinction of methods and properties in Powershell and traditional programming languages and how they can interact with Get-Member to solve the koans.

## Changes
- explain properties in OOP 
- explain methods in OOP
- show example usage on how and when to use each
- explain how Get-Member can help when using the membertypes Property and Method

## Checklist

- [x] Pull Request has a meaningful title.
- [x] Summarised changes.
- [x] Pull Request is ready to merge & is not WIP.
- [ ] Added tests / only testable interactively.
  - Make sure you add a new test if old tests do not effectively test the code changed.
- [ ] Added documentation / opened issue to track adding documentation at a later date.

### CODE REVIEW

File: PSKoans/Koans/Introduction/AboutGetMember.Koans.ps1 [hunk 1]
```diff
@@ -166,7 +182,7 @@ Describe "Get Member" {
 
             $Reason = $MethodString -f $MethodName, $CmdletName
             & (Get-Command -Name $CmdletName) |
-                Get-Member -MemberType Property -Name $MethodName |
+                Get-Member -MemberType Method -Name $MethodName |
```
Comment (vexx32) [inline PSKoans/Koans/Introduction/AboutGetMember.Koans.ps1:185]: Yikes, good catch there.

**Review by vexx32 — CHANGES_REQUESTED** [review]
Looks lovely overall! A few minor comments to look at, but thanks for this one! 💖

File: PSKoans/Koans/Cmdlets 1/AboutDiscovery.Koans.ps1 [hunk 2]
```diff
@@ -103,10 +134,22 @@ Describe 'Get-Member' {
 
             # Methods can be accessed just like properties, but have parentheses and often parameters!
             $String.EndsWith('____') | Should -BeTrue
+
             <#
+                Contrary to properties that represent the state of an object, methods perform actions
+                on the object. Thus, they sometimes need additional information as opposed to
+                properties you can query as is. This is where the parentheses come into play.
+                Inside the parentheses you give additional parameters (oftentimes called arguments)
+                to the method, similary to parameters for a function. In fact, this is how most
```
[the remark this answers is not in the document]
Comment (Maamue) [reply PSKoans/Koans/Cmdlets 1/AboutDiscovery.Koans.ps1:143]: Typo + not a native speaker, wanted to use "similarly" originally :)

File: PSKoans/Koans/Introduction/AboutGetMember.Koans.ps1 [hunk 3]
```diff
@@ -43,6 +43,13 @@ Describe "Get Member" {
         <#
             Let's look at some object properties!
 
+            Properties define the state an object is in. In traditional object-
+            oriented programming languages, a state of an objet could only be changed
+            by calling provided methods on the object itself. Powershell is more lenient
```
[the remark this answers is not in the document]
Comment (Maamue) [reply PSKoans/Koans/Introduction/AboutGetMember.Koans.ps1:48]: Completely rewrote this bit (removed it, to be exact). Wanted to explain a bit how the concept of methods and properties are used in a general OOP sense, but that is not necessary for the koans.

File: PSKoans/Koans/Cmdlets 1/AboutDiscovery.Koans.ps1 [hunk 4]
```diff
@@ -103,10 +134,22 @@ Describe 'Get-Member' {
 
             # Methods can be accessed just like properties, but have parentheses and often parameters!
             $String.EndsWith('____') | Should -BeTrue
+
             <#
+                Contrary to properties that represent the state of an object, methods perform actions
+                on the object. Thus, they sometimes need additional information as opposed to
+                properties you can query as is. This is where the parentheses come into play.
```
[the remark this answers is not in the document]
Comment (Maamue) [reply PSKoans/Koans/Cmdlets 1/AboutDiscovery.Koans.ps1:141]: Removed this bit from the other topic, but summarised it anyway. Makes it easier to understand the actual topic of the koan.

File: PSKoans/Koans/Introduction/AboutGetMember.Koans.ps1 (hunk 1 above)
Comment (Maamue) [reply PSKoans/Koans/Introduction/AboutGetMember.Koans.ps1:185]: Yeah, I was confused for a bit if I did something wrong until it hit me.

**Review by vexx32 — APPROVED** [review]

**Pull request merged**